components/function-controller/pkg/utils/equality.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 The Kyma Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+
+	servingv1alpha1 "knative.dev/serving/pkg/apis/serving/v1alpha1"
+)
+
+// Semantic can do semantic deep equality checks for API objects.
+// Fields which are not relevant for Model-T are intentionally omitted.
+var Semantic = conversion.EqualitiesOrDie(
+	ksvcEqual,
+)
+
+// ksvcEqual asserts the equality of two Knative Service objects.
+func ksvcEqual(s1, s2 *servingv1alpha1.Service) bool {
```

Comment:
i see no handling of the RouteSpec that you can put into a ksvc. is this on purpose?

Reply:
It's on purpose, because AFAIK we don't fiddle with it. I don't mind adding it for completion if you think we should care.

Reply:
lets put it in when we need it